lib/update/cluster/builder.go

```diff
@@ -637,16 +637,22 @@ func supportsTaints(gravityPackage loc.Locator) (supports bool, err error) {
 	return defaults.BaseTaintsVersion.Compare(*ver) <= 0, nil
 }
 
-func shouldUpdateEtcd(installedRuntimePackage, updateRuntimePackage loc.Locator, packages pack.PackageService) (etcd *etcdVersion, err error) {
+func shouldUpdateEtcd(installedRuntimePackage, updateRuntimePackage, altInstalledRuntimePackage loc.Locator, packages pack.PackageService) (etcd *etcdVersion, err error) {
 	// TODO: should somehow maintain etcd version invariant across runtime packages
 	var updateEtcd bool
 	installedVersion, err := getEtcdVersion("version-etcd", installedRuntimePackage, packages)
 	if err != nil {
-		if !trace.IsNotFound(err) {
+		// Workaround for https://github.com/gravitational/gravity/issues/2031
+		// If we loaded a corrupted version of the planet version information, fallback to the default runtime package
+		// on versions where this was included.
+		// TODO(knisbet): Obsolete when no longer supporting upgrades from 7.0
+		if strings.Contains(err.Error(), "REPLACE_ETCD_LATEST_VERSION") && !altInstalledRuntimePackage.IsEmpty() {
```

Comment:
Wouldn't it be more reliable to just always fall back to the default runtime package?

Reply:
Probably, my thought is that would potential hide unknown forms of corruption. So the narrow approach allows us to target the bug we know about, and avoid masking related problems we might not know about (WDKWWDK).